libs/tree-sitter-wing/grammar.js

```diff
@@ -24,44 +24,54 @@ module.exports = grammar({
 
   word: ($) => $.identifier,
 
+  externals: ($) => [$.AUTOMATIC_SEMICOLON],
+
   precedences: ($) => [
     // Handle ambiguity in case of empty literal: `a = {}`
     // In this case tree-sitter doesn't know if it's a set or a map literal so just assume its a map
     [$.map_literal, $.set_literal],
     [$.json_literal, $.structured_access_expression],
   ],
 
-  conflicts: ($) => [
-    [$.expression, $.custom_type]
-  ],
+  conflicts: ($) => [[$._reference_identifier, $._type_identifier]],
 
   supertypes: ($) => [$.expression, $._literal],
 
   rules: {
     // Basics
     source: ($) => repeat($._statement),
     block: ($) => seq("{", optional(repeat($._statement)), "}"),
+    _semicolon: ($) => choice(";", $.AUTOMATIC_SEMICOLON),
```

Comment:
Curious, why `_`? (feels unnecessarily apologetic)

Reply:
_ makes it hidden, which is nice because our AST really doesn't need to show semicolons.
Luckily though, when the external scanner "inserts" a semicolon the `AUTOMATIC_SEMICOLON` node _does_ show up in the AST. So it's kinda a win-win